### ISSUE TITLE
pod/perldata.pod: Clarify discussion of identifier parsing

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#13078]
Migrated from [rt.perl.org#118723](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=118723) (status was 'resolved')

Searchable as RT118723$

**Comment by p5pRT** [comment]
### From @jkeenan

There is a very confusing paragraph in perldata's section on identifier 
parsing\.  Here is a patch for it; please review\.

Thank you very much\.
Jim Keenan

**Comment by p5pRT** [comment]
### From @jkeenan

<details><summary>0001-Delete-superfluous-lines-clarify-identifier-parsing.patch</summary>

```diff
From 9db514fb0466d6985ddae80a3acfe5886ffb2a04 Mon Sep 17 00:00:00 2001
From: James E Keenan <jkeenan@cpan.org>
Date: Wed, 3 Jul 2013 01:20:23 +0200
Subject: [PATCH] Delete superfluous lines; clarify identifier parsing.

---
 pod/perldata.pod |    7 +------
 1 files changed, 1 insertions(+), 6 deletions(-)

diff --git a/pod/perldata.pod b/pod/perldata.pod
index 235177b..1cf2e1f 100644
--- a/pod/perldata.pod
+++ b/pod/perldata.pod
@@ -143,12 +143,7 @@ Single quotes have similar rules, but with the exception that they are not
 legal at the end of an identifier: That is, C<$'foo> and C<$foo'bar> are
 legal, but C<$foo'bar'> are not.
 
-
-Finally, if the identifier is preceded by a sigil --
-More so, normal identifiers can start or end with any number
-of double colons (::), and can contain several parts delimited
-by double colons.
-And additionally, if the identifier is preceded by a sigil --
+Additionally, if the identifier is preceded by a sigil --
 that is, if the identifier is part of a variable name -- it
 may optionally be enclosed in braces.
 
-- 
1.7.1


```
</details>

**Comment by p5pRT** [comment]
### From @jkeenan

$ perl \-V
Summary of my perl5 \(revision 5 version 18 subversion 0\) configuration&#8203;:
&nbsp;   
&nbsp;  Platform&#8203;:
&nbsp;    osname=darwin\, osvers=8\.11\.0\, archname=darwin\-2level
&nbsp;    uname='darwin macintosh\-9\.local 8\.11\.0 darwin kernel version 8\.11\.0&#8203;: wed oct 10 18&#8203;:26&#8203;:00 pdt 2007; root&#8203;:xnu\-792\.24\.17~1release\_ppc power macintosh powerpc '
&nbsp;    config\_args='\-des'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=undef\, usemultiplicity=undef
&nbsp;    useperlio=define\, d\_sfio=undef\, uselargefiles=define\, usesocks=undef
&nbsp;    use64bitint=undef\, use64bitall=undef\, uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, ccflags ='\-fno\-common \-DPERL\_DARWIN \-fno\-strict\-aliasing \-pipe \-I/usr/local/include \-I/opt/local/include'\,
&nbsp;    optimize='\-O3'\,
&nbsp;    cppflags='\-fno\-common \-DPERL\_DARWIN \-fno\-strict\-aliasing \-pipe \-I/usr/local/include \-I/opt/local/include'
&nbsp;    ccversion=''\, gccversion='4\.0\.1 \(Apple Computer\, Inc\. build 5250\)'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=4321
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='env MACOSX\_DEPLOYMENT\_TARGET=10\.3 cc'\, ldflags =' \-L/usr/local/lib \-L/opt/local/lib'
&nbsp;    libpth=/usr/local/lib /opt/local/lib /usr/lib
&nbsp;    libs=\-ldbm \-ldl \-lm \-lc
&nbsp;    perllibs=\-ldl \-lm \-lc
&nbsp;    libc=\, so=dylib\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version=''
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=bundle\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;    cccdlflags=' '\, lddlflags=' \-bundle \-undefined dynamic\_lookup \-L/usr/local/lib \-L/opt/local/lib'


Characteristics of this binary \(from libperl\)&#8203;: 
&nbsp;  Compile\-time options&#8203;: HAS\_TIMES PERLIO\_LAYERS PERL\_DONT\_CREATE\_GVSV
&nbsp;                        PERL\_HASH\_FUNC\_ONE\_AT\_A\_TIME\_HARD PERL\_MALLOC\_WRAP
&nbsp;                        PERL\_PRESERVE\_IVUV PERL\_SAWAMPERSAND USE\_LARGE\_FILES
&nbsp;                        USE\_LOCALE USE\_LOCALE\_COLLATE USE\_LOCALE\_CTYPE
&nbsp;                        USE\_LOCALE\_NUMERIC USE\_PERLIO USE\_PERL\_ATOF
&nbsp;  Built under darwin
&nbsp;  Compiled at May 18 2013 12&#8203;:31&#8203;:26
&nbsp;  %ENV&#8203;:
&nbsp;    PERLBREW\_BASHRC\_VERSION="0\.59"
&nbsp;    PERLBREW\_HOME="/Users/jimk/\.perlbrew"
&nbsp;    PERLBREW\_MANPATH=""
&nbsp;    PERLBREW\_PATH="/Users/jimk/perl5/perlbrew/bin"
&nbsp;    PERLBREW\_ROOT="/Users/jimk/perl5/perlbrew"
&nbsp;    PERLBREW\_VERSION="0\.59"
&nbsp;  @&#8203;INC&#8203;:
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.18\.0/darwin\-2level
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.18\.0
&nbsp;    /usr/local/lib/perl5/5\.18\.0/darwin\-2level
&nbsp;    /usr/local/lib/perl5/5\.18\.0
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.16\.0
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.14\.2
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.14\.0
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.12\.0
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.10\.1
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.10\.0
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.8\.6
&nbsp;    /usr/local/lib/perl5/site\_perl
&nbsp;    \.

**Comment by p5pRT** [comment]
### From @cpansprout

On Tue Jul 02 16&#8203;:27&#8203;:58 2013\, jkeen@&#8203;verizon\.net wrote&#8203;:
> There is a very confusing paragraph in perldata's section on identifier 
> parsing\.  Here is a patch for it; please review\.
> 
> Thank you very much\.
> Jim Keenan
> 

Sounds good\.  Can you change ‘are’ to ‘is’ on the line above\, while you
are at it?

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @jkeenan

On Tue Jul 02 17&#8203;:58&#8203;:26 2013\, sprout wrote&#8203;:
> On Tue Jul 02 16&#8203;:27&#8203;:58 2013\, jkeen@&#8203;verizon\.net wrote&#8203;:
> > There is a very confusing paragraph in perldata's section on identifier 
> > parsing\.  Here is a patch for it; please review\.
> > 
> > Thank you very much\.
> > Jim Keenan
> > 
> 
> Sounds good\.  Can you change ‘are’ to ‘is’ on the line above\, while you
> are at it?
> 

Done\.  Thanks for the review\.  Merged in commit 1d26800\.  Resolving ticket\.

**Comment by p5pRT** [comment]
@jkeenan - Status changed from 'open' to 'resolved'
